src/pages/index.js

```diff
@@ -25,37 +26,115 @@ editProfileValidator.enableValidation();
 const addCardValidator = new FormValidator(config, newCardEditForm);
 addCardValidator.enableValidation();
 
+const avatarFormValidator = new FormValidator(config, avatarForm);
+avatarFormValidator.enableValidation();
+
+const removeCardValidator = new FormValidator(config, removeCardForm);
+removeCardValidator.enableValidation();
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "eeca6a67-c0e1-4af7-ac48-d61d2e5d5520",
+    "Content-Type": "application/json",
+  },
+});
+
+let cardSection;
+
+api.getInitialCards().then((cards) => {
+  cardSection = new Section(
+    {
+      items: cards,
+      renderer: (cardData) => {
+        const cardElement = getCardElement(cardData);
+        cardSection.addItem(cardElement);
+      },
+    },
+    ".cards__list"
+  );
+  cardSection.renderItems();
+});
+const userInfo = new UserInfo({
+  userName: "#profile-title-js",
+  userAbout: "#profile-description-js",
+  userAvatar: ".profile__image",
+});
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo(userData);
+});
+
 /**================================================================================================
  *                                         FUNCTION
  *================================================================================================**/
 
 function getCardElement(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleCardDelete,
+    handleLikeClick
+  );
   return card.getView();
 }
 /**------------------------------------------------------------------------------------------------
  *                                         EVENT HANDLERS
  *------------------------------------------------------------------------------------------------**/
 
-function handleProfileFormSubmit(formData) {
-  userInfo.setUserInfo({ name: formData.title, job: formData.description });
-  profileEditPopup.close();
-}
-
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileFormSubmit
+  (formData) => {
+    profileEditPopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```